stdlib/src/utils/static_tuple.mojo

```diff
@@ -97,6 +97,70 @@ fn _create_array[
         return array
 
 
+@always_inline
+fn _set_array_elem_mem[
+    index: Int,
+    size: Int,
+    type: CollectionElement,
+](
+    val: type,
+    array: Reference[
+        __mlir_type[`!pop.array<`, size.value, `, `, type, `>`],
+        __mlir_attr.`1 : i1`,
+        _,
+    ],
+):
+    """Sets the array element at position `index` with the value `val`.
+
+    Parameters:
+        index: the position to replace the value at.
+        size: the size of the array.
+        type: the element type of the array
+
+    Args:
+        val: the value to set.
+        array: the array which is captured by reference.
+    """
+    var ptr = __mlir_op.`pop.array.gep`(
+        array.get_legacy_pointer().address, index.value
+    )
+    var p = UnsafePointer(ptr)
+    initialize_pointee(p, val)
+
+
+@always_inline
+fn _create_array_mem[
+    size: Int, type: CollectionElement
+](lst: VariadicList[type]) -> __mlir_type[
```

Comment:
**Question** I think I had an earlier comment around using `VariadicList`, but I think we want this to be `*lst: type` to mean "a variadic pack of types" rather than a `VariadicList`.  In general, you don't want to spell `VariadicList` or `VariadicListMem` directly in APIs.

Reply:
See https://github.com/modularml/mojo/pull/2294#discussion_r1569499182